src/test/java/org/alfresco/repo/rawevents/EventBehaviorTest.java

```diff
@@ -0,0 +1,421 @@
+
+        MockEndpoint mockEndpoint = camelContext.getEndpoint(endpointUri, MockEndpoint.class);
+        mockEndpoint.setAssertPeriod(500);
+        mockEndpoint.setExpectedCount(1);
+
+        BehaviourDefinition<ClassBehaviourBinding> classBehaviour = null;
+        try
+        {
+            // start the transaction
+            txn = getUserTransaction();
+            txn.begin();
+
+            setupTestData();
+
+            EventBehaviour eventBehaviour = new EventBehaviour(eventProducer, endpointUri, this, "onContentUpdateBehaviourTest", Behaviour.NotificationFrequency.EVERY_EVENT);
+
+            // Register interest in the content update event for a versionable node
+            classBehaviour = this.policyComponent.bindClassBehaviour(ContentServicePolicies.OnContentUpdatePolicy.QNAME, ContentModel.ASPECT_VERSIONABLE, eventBehaviour);
+
+            // First check that the policy is not fired when the versionable aspect is not
+            // present
+            ContentWriter contentWriter = this.contentService.getWriter(contentNodeRef, ContentModel.PROP_CONTENT, true);
+            contentWriter.putContent("content update one");
+            assertFalse(this.policyFired);
+
+            this.newContent = false;
+
+            // Now check that the policy is fired when the versionable aspect is present
+            this.nodeService.addAspect(this.contentNodeRef, ContentModel.ASPECT_VERSIONABLE, null);
+            ContentWriter contentWriter2 = this.contentService.getWriter(contentNodeRef, ContentModel.PROP_CONTENT, true);
+            contentWriter2.putContent("content update two");
+            assertTrue(this.policyFired);
+            this.policyFired = false;
+
+            // Check that the policy is not fired when using a non updating content writer
+            ContentWriter contentWriter3 = this.contentService.getWriter(contentNodeRef, ContentModel.PROP_CONTENT, false);
+            contentWriter3.putContent("content update three");
+            assertFalse(this.policyFired);
+
+            try
+            {
+                eventBehaviour.disable();
+
+                ContentWriter contentWriter4 = this.contentService.getWriter(contentNodeRef, ContentModel.PROP_CONTENT, true);
+                contentWriter4.putContent("content update four");
+                assertFalse(this.policyFired);
+            }
+            finally
+            {
+                eventBehaviour.enable();
+            }
+
+            // Assert that the endpoint received 1 message
+            mockEndpoint.assertIsSatisfied();
+        }
+        finally
+        {
+            if (txn != null)
+            {
+                txn.rollback();
+            }
+
+            if (classBehaviour != null)
+            {
+                this.policyComponent.removeClassDefinition(classBehaviour);
+            }
+        }
+
+    }
+
+    /**
+     * Tests that the content update policy is triggered correctly for every event.
+     */
+    // @Test
```

Comment:
We can spin up ActiveMQ in Docker with PostreSQL for tests.
I can see it as a separate Job in Bamboo.

Reply:
Re-enabled the test. We will need to update the bamboo job.